### PR TITLE
Avoid prefixing columns when `false` is assigned to `column-prefix`

### PR DESCRIPTION
**Pull request by andreyvital**
http://www.doctrine-project.org/jira/browse/DDC-3293

As the issue on JIRA wasn't updated since September/2014, here's a PR.

@Ocramius suggested on [issue](http://www.doctrine-project.org/jira/browse/DDC-3293) about introducing a new attribute called `use-column-prefix`, I'd say that keeping the same attribute as in others drivers would be better.

That's it.

### CODE REVIEW

**Comment by Ocramius** [comment]
Looking good: :+1: 

@CentaurWarchief as usual, I have to ask for tests.

File: lib/Doctrine/ORM/Mapping/Driver/XmlDriver.php [hunk 1]
```diff
@@ -833,6 +842,6 @@ protected function evaluateBoolean($element)
     {
         $flag = (string)$element;
 
-        return ($flag === true || $flag == "true" || $flag == "1");
```
Comment (Ocramius) [inline lib/Doctrine/ORM/Mapping/Driver/XmlDriver.php]: Why did you remove the simple boolean `true` case?

File: lib/Doctrine/ORM/Mapping/Driver/XmlDriver.php [hunk 2]
```diff
@@ -255,11 +255,20 @@ public function loadMetadataForClass($className, ClassMetadata $metadata)
 
         if (isset($xmlRoot->embedded)) {
             foreach ($xmlRoot->embedded as $embeddedMapping) {
+                $columnPrefix = isset($embeddedMapping['column-prefix'])
+                    ? (string) $embeddedMapping['column-prefix']
+                    : null;
+
+                $preventPrefixing = (
```
Comment (Ocramius) [inline lib/Doctrine/ORM/Mapping/Driver/XmlDriver.php]: `$preventPrefixing = $this->evaluateBoolean($embeddedMapping['column-prefix'])`

File: lib/Doctrine/ORM/Mapping/Driver/XmlDriver.php [hunk 3]
```diff
@@ -255,11 +255,20 @@ public function loadMetadataForClass($className, ClassMetadata $metadata)
 
         if (isset($xmlRoot->embedded)) {
             foreach ($xmlRoot->embedded as $embeddedMapping) {
+                $columnPrefix = isset($embeddedMapping['column-prefix'])
```
Comment (Ocramius) [inline lib/Doctrine/ORM/Mapping/Driver/XmlDriver.php]: Actually, this fix is incorrect, now that I look at it. See description of issue http://www.doctrine-project.org/jira/browse/DDC-3293 - I used an additional property there, and that is because `true` and `false` are still valid string prefixes.

File: lib/Doctrine/ORM/Mapping/Driver/XmlDriver.php (hunk 1 above)
Comment (CentaurWarchief) [inline lib/Doctrine/ORM/Mapping/Driver/XmlDriver.php]: Because `(string) $element` never will be true. They're different types.

```
$ php -r 'var_dump((string) 'true' === true);' // bool(false)
```

Comment (Ocramius) [inline lib/Doctrine/ORM/Mapping/Driver/XmlDriver.php]: Good point

File: lib/Doctrine/ORM/Mapping/Driver/XmlDriver.php (hunk 2 above)
Comment (CentaurWarchief) [inline lib/Doctrine/ORM/Mapping/Driver/XmlDriver.php]: Wrong. It will return true for a valid prefix. I only need ensure that false or 0 was given.

File: lib/Doctrine/ORM/Mapping/Driver/XmlDriver.php (hunk 3 above)
Comment (CentaurWarchief) [inline lib/Doctrine/ORM/Mapping/Driver/XmlDriver.php]: I've said before, I'm not making use of a additional attribute as you've proposed.

I'd really think that there's no need for adding another attribute, well, we only need ensure if `false` or `0` was given for `column-prefix`.

It will be always string, of course, we can't change it, but, it's only input treatment for us, if false was given, so we understand the `false` as that column should not be prefixed. The only difference is that any incoming value from a XML attribute will be always string :)

Comment (Ocramius) [inline lib/Doctrine/ORM/Mapping/Driver/XmlDriver.php]: > The only difference compared to `YML`, etc... is that any incoming value from a XML attribute will be always string :) 

That's indeed why you need an additional attribute.

Comment (CentaurWarchief) [inline lib/Doctrine/ORM/Mapping/Driver/XmlDriver.php]: I'd say you're correct, I'm thinking better now since my last comment lol, makes sense.

File: lib/Doctrine/ORM/Mapping/Driver/XmlDriver.php (hunk 2 above)
Comment (CentaurWarchief) [inline lib/Doctrine/ORM/Mapping/Driver/XmlDriver.php]: :100:

**Comment by Ocramius** [comment]
@CentaurWarchief merged, thanks!
